### PR TITLE
Make cleanup check before deleting

### PR DESCRIPTION
**Pull request by daveh86**
Not ready to merge yet

### CODE REVIEW

**Comment by daveh86** [comment]
@agorrod, would you be able to take a look at this? Its a small change to check that a directory exists before we attempt to delete it. This fixes the issues we are currently seeing on the Windows tester

File: test/utility/test_util.i [hunk 1]
```diff
@@ -108,8 +111,11 @@ testutil_clean_work_dir(char *dir)
 		testutil_die(ENOMEM, "Failed to allocate memory");
 
 	snprintf(buffer, inputSize, "%s%s", RM_COMMAND, dir);
-	if ((ret = system(buffer)) != 0)
-		testutil_die(ret, "System call to remove directory failed");
+
+	if ( stat( dir, &info ) == 0 )
```
Comment (agorrod) [inline test/utility/test_util.i]: The spacing still isn't right here. There should not be a space before ```stat``` or after the ```0```.

**Comment by agorrod** [comment]
@daveh86 I'd rather use the ```___wt_exists``` API here (see ```src/os_posix/os_exist.c```).

It's a platform independent version of the functionality. The trick is that it converts filenames to WiredTiger specific names if they aren't absolute, so it will only work in this situation if you pass in an absolute path.

File: test/thread/thread.h [hunk 2]
```diff
@@ -38,8 +38,7 @@
 #include <string.h>
 #include <unistd.h>
 
-#include "wt_internal.h"			/* For __wt_XXX */
-#include "test_util.i"
+	#include "test_util.i"
```
Comment (agorrod) [inline test/thread/thread.h]: Extra spaces at the start of the line.

File: test/utility/test_util.i [hunk 3]
```diff
@@ -28,7 +28,10 @@
 #include <string.h>
 #include <stdlib.h>
 #include <stdio.h>
+#include <sys/types.h>
+#include <sys/stat.h>
```
Comment (agorrod) [inline test/utility/test_util.i]: Are these includes still needed?

File: test/utility/test_util.i [hunk 4]
```diff
@@ -99,7 +102,7 @@ static inline void
 testutil_clean_work_dir(char *dir)
 {
 	size_t inputSize;
-	int ret;
+	int ret, existp;
```
Comment (agorrod) [inline test/utility/test_util.i]: Order should be switched, so they are alphabetical.

File: test/utility/test_util.i [hunk 5]
```diff
@@ -108,8 +111,15 @@ testutil_clean_work_dir(char *dir)
 		testutil_die(ENOMEM, "Failed to allocate memory");
 
 	snprintf(buffer, inputSize, "%s%s", RM_COMMAND, dir);
-	if ((ret = system(buffer)) != 0)
-		testutil_die(ret, "System call to remove directory failed");
+
+	existp = 0;
```
Comment (agorrod) [inline test/utility/test_util.i]: This variable should be ```exist```, the ```p``` at the end indicates a pointer - and the local variable isn't a pointer.

File: test/utility/test_util.i [hunk 6]
```diff
@@ -108,8 +111,15 @@ testutil_clean_work_dir(char *dir)
...
+	if ( __wt_exist(NULL, dir, &existp) != 0 )
+		testutil_die(existp,
+			    "Unable to check if directory exists");
```
Comment (agorrod) [inline test/utility/test_util.i]: This indent looks wrong - it should be 4 spaces.

File: test/utility/test_util.i [hunk 7]
```diff
@@ -108,8 +111,15 @@ testutil_clean_work_dir(char *dir)
 		testutil_die(ENOMEM, "Failed to allocate memory");
 
 	snprintf(buffer, inputSize, "%s%s", RM_COMMAND, dir);
-	if ((ret = system(buffer)) != 0)
-		testutil_die(ret, "System call to remove directory failed");
+
+	existp = 0;
+	if ( __wt_exist(NULL, dir, &existp) != 0 )
```
Comment (agorrod) [inline test/utility/test_util.i]: There is a trailing space after the ```0```

File: test/utility/test_util.i [hunk 8]
```diff
@@ -108,8 +111,15 @@ testutil_clean_work_dir(char *dir)
...
+		testutil_die(existp,
+			    "Unable to check if directory exists");
+	if (existp == 1)
+		if ((ret = system(buffer)) != 0)
```
Comment (agorrod) [inline test/utility/test_util.i]: You could change this to a single conditional:

```if (existp == 1 && (ret = system(buffer)) != 0)```

File: test/utility/test_util.i (hunk 7 above)
Comment (agorrod) [inline test/utility/test_util.i]: It would also be normal to save the return value into ```ret```, and use it in the error message.

**Comment by agorrod** [comment]
Thanks @daveh86 lgtm - I'll merge

**Pull request merged**